File: Data Structures & Algorithms/binary-search/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums)-1
        while left <= right:
            m = left + (right - left)//2

            if nums[m] == target:
                return m
            if nums[m] < target:
                left = m +1
            else:
                right = m - 1
                
        return -1

File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer, Integer> map = new HashMap<>();
        int [] results = new int[2];
        for(int i = 0; i < nums.length; i++){
            int get_val = map.getOrDefault(target-nums[i], -1);
            if(get_val != -1){
                results[0] = map.get(target-nums[i]);
                results[1] = i;
                return results;
            } else{
                map.put(nums[i], i);
            }
        }
        return results;
    }
}
